jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.s3
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import org.jetbrains.annotations.TestOnly
+import software.amazon.awssdk.services.s3.S3Client
+import software.aws.toolkits.resources.message
+import java.awt.Component
+import javax.swing.JComponent
+
+class CreateS3BucketDialog(
+    private val project: Project,
+    private val s3Client: S3Client,
+    private val parent: Component? = null
+) : DialogWrapper(project, parent, false, IdeModalityType.PROJECT) {
+
+    private val view = CreateBucketPanel(project)
+
+    var bucket: String? = null
```

Comment:
Why do we need the `bucket` instance variable, the `bucketName()` method and the access from the testfield ?

Reckon we can just fetch it straight from the text field.... no need to have two mutable states.

Reply:
I was thinking that `bucketName` always returns whatever is in the text field whether is validated or not. `bucket` returns the validated value which is safe to use outside of the Panel.

Reply:
Removed the `bucket` field as it doesn't gain too much. Using `bucketName()` instead